Cargill/splinter
# Add comments field to Circuit

Add the `comments` field to the `Circuit` protobuf and the
corresponding native representation, `CreateCircuit`. This `string`
field can be used to provide human-readable comments about the proposed
circuit to make it easier for admins to distinguish between proposed
circuits.

Also adds the `--comments` argument for the circuit create CLI to set this field, and shows these comments when listing proposals in the CLI.

To test:
1. Run gameroom: `CARGO_ARGS="-- --features=experimental" docker-compose -f examples/gameroom/docker-compose.yaml up --build`
2. Open a session in the `generate-key-registry` container: `docker-compose -f examples/gameroom/docker-compose.yaml run generate-key-registry bash`
3. Submit a circuit proposal with a comment:
 ```
 splinter circuit create \
   --key /key_registry/alice.priv \
   --url http://splinterd-node-acme:8085  \
   --node acme-node-000::tls://splinterd-node-acme:8044 \
   --node bubba-node-000::tls://splinterd-node-bubba:8044 \
   --service sc00::acme-node-000 \
   --service sc01::bubba-node-000 \
   --service-type *::scabbard \
   --management custom \
   --service-arg *::admin_keys=$(cat /key_registry/alice.pub) \
   --service-peer-group sc00,sc01 \
   --comments "Something useful"
 ```
4. Check that the comment shows up in the proposals list: `splinter circuit proposals --url http://splinterd-node-bubba:8085`

You can also verify that this change is backwards compatible by starting up gameroom from master, creating a proposal, then restarting gameroom with this branch and seeing that it starts up without errors.
